File: test/tree/lowest-common-ancestor-of-a-binary-tree.test.ts
import { array2tree } from '../../src/tree/common/tree-node';
import { lowestCommonAncestor } from '../../src/tree/lowest-common-ancestor-of-a-binary-tree';

describe('lowest common ancestor of a binary tree i', () => {
  test('lowest common ancestor - test case 1', async () => {
    const tree = array2tree([3, 5, 1, 6, 2, 0, 8, null, null, 7, 4]);

    expect(lowestCommonAncestor(tree, tree!.left, tree!.right)?.val).toStrictEqual(3);
  });
});

File: test/leetcode/tree/lowest_common_ancestor_of_a_binary_tree_test.py
from leetcode.tree.common.tree_node import array2tree
from leetcode.tree.lowest_common_ancestor_of_a_binary_tree import Solution


soln = Solution()


def test_case_1():
    tree = array2tree([3, 5, 1, 6, 2, 0, 8, None, None, 7, 4])
    assert tree
    assert tree.left
    assert tree.right

    lca = soln.lowestCommonAncestor(tree, tree.left, tree.right)
    assert lca

    assert lca.val == 3
